packages/calypso-products/src/is-difm-product.ts

```diff
@@ -0,0 +1,8 @@
+import { WPCOM_DIFM_LITE } from './constants';
+import { formatProduct } from './format-product';
+
+export function isDIFMProduct( product: any ): boolean {
+	product = formatProduct( product );
```

Comment:
This is _probably_ not necessary. It changes some of the properties in `product` to `snake_case`. You only need it if you are calling this function with data that's not correct, but in this case you're using the properties from a cart, which will always be correct. I think we should stop using it in all these functions to make it more clear what data we have.

Reply:
Yeah but can we be sure the cart items will never be snake case? Asking because I am uncertain as well :)

Reply:
These functions were created ages ago, long before we had TypeScript and also when the shopping-cart system was not well documented. The reason they support both camelCase and snake_case is that they were (and still are in some cases) used for _different types_ of data (eg: shopping-cart products vs. products endpoint products, etc.). This really isn't ideal. For example, there are functions in this directory that use `formatProduct` and then proceed to look for a product property that doesn't even get changed by that function. When types are unclear, we're left with superstition. It's harmless to do, but if the caller of this function doesn't know the type of data they're dealing with, they're likely to leave themselves open to subtle bugs.

Eventually, we may even want to remove these functions entirely (this was suggested by the Calypso team and I agree) as they bloat our bundles a lot and don't provide much benefit. Rather than a function that tries to guess what property you have and compare that property's slug, we can instead just compare the slug ourselves (which should be easy if we know what type of data we have).